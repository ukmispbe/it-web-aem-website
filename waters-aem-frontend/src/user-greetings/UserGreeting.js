import React from 'react';
import SessionStore from '../stores/sessionStore';

function UserGreeting(props) {
    const { greetings, logoDirectoryPath } = props;
    const store = new SessionStore();
    const savedUserDetails = store.getUserDetails();
    const name = `${savedUserDetails.firstName || ''} ${savedUserDetails.lastName || ''}`;
    const company = savedUserDetails.company || '';
    const [isImageHidden, hideCompanyImageOnError] = useState(false);
    return (
        <>
            <div className="greetings" data-locator="user-greeting-sec">
                <h2 data-locator="user-greeting">{greetings}</h2>
                <h3 data-locator="user-greeting-name">{name.trim()}</h3>
                <h4 data-locator="user-greeting-company">{company}</h4>
            </div>
            {!isImageHidden && <img
                src={`${logoDirectoryPath.replace(/\/$/, '')}/${company.trim().replace(/ /g, '-').toLowerCase()}.png`}
                alt={company}
                className="logo"
                data-locator="user-greeting-logo"
                onError={() => hideCompanyImageOnError(true)}
            />}
        </>
    );
}

UserGreeting.defaultProps = {
    greetings: '',
    logoDirectoryPath: ''
};

export default UserGreeting;