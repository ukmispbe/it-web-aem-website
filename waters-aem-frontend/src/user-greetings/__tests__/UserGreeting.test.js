import React from 'react';
import { shallow } from 'enzyme';

//Mocked Props
import UserGreeting from '../UserGreeting';
import props, { userDetails } from '../UserGreeting.mock';

describe('<UserGreeting />', () => {
    let enzymeWrapper;
    beforeAll(() => {
        window.sessionStorage = {
            storage: {},
            setItem: function (field, value) {
                this.storage[field] = value;
            },
            getItem: function (field) {
                return value ? JSON.parse(this.storage[field]) : {};
            }
        }
        window.sessionStorage.setItem('waters.userDetails', JSON.stringify(userDetails));

        enzymeWrapper = shallow(<UserGreeting {...props} />);
    });
    it('should render without throwing an error', () => {
        expect(enzymeWrapper).toMatchSnapshot();
    });
    it('should display greeting, name, company, and logo', () => {
        expect(enzymeWrapper.find('.greetings')).toHaveLength(1);
        expect(enzymeWrapper.find('h2')).toHaveLength(1);
        expect(enzymeWrapper.find('h3')).toHaveLength(1);
        expect(enzymeWrapper.find('h4')).toHaveLength(1);
        expect(enzymeWrapper.find('.logo')).toHaveLength(1);

        expect(enzymeWrapper.find('h2').text()).toBe(props.greetings);
        expect(enzymeWrapper.find('h3').text()).toBe(`${userDetails.firstName} ${userDetails.lastName}`);
        expect(enzymeWrapper.find('h4').text()).toBe(userDetails.company);
    });
    it('should display greeting, firstname, company, and logo', () => {
        window.sessionStorage.setItem('waters.userDetails', JSON.stringify({ ...userDetails, lastName: '' }));
        enzymeWrapper.setProps({ greetings: 'Welcome' });
        enzymeWrapper.update();
        expect(enzymeWrapper.find('h2')).toHaveLength(1);
        expect(enzymeWrapper.find('h3')).toHaveLength(1);
        expect(enzymeWrapper.find('h4')).toHaveLength(1);
        expect(enzymeWrapper.find('.logo')).toHaveLength(1);

        expect(enzymeWrapper.find('h2').text()).toBe('Welcome');
        expect(enzymeWrapper.find('h3').text()).toBe(userDetails.firstName);
        expect(enzymeWrapper.find('h4').text()).toBe(userDetails.company);
    });
    it('should display greeting, lastname and no company no logo', () => {
        window.sessionStorage.setItem('waters.userDetails', JSON.stringify({ ...userDetails, firstName: '', company: '' }));
        enzymeWrapper.setProps({ greetings: 'Welcome' });
        enzymeWrapper.update();
        expect(enzymeWrapper.find('h2')).toHaveLength(1);
        expect(enzymeWrapper.find('h3')).toHaveLength(1);
        expect(enzymeWrapper.find('h4')).toHaveLength(1);
        expect(enzymeWrapper.find('.logo')).toHaveLength(1);

        expect(enzymeWrapper.find('h2').text()).toBe('Welcome');
        expect(enzymeWrapper.find('h3').text()).toBe(userDetails.lastName);
        expect(enzymeWrapper.find('h4').text()).not.toBe(userDetails.company);
    });
    it('should display greeting, no name, no company and no logo', () => {
        window.sessionStorage.setItem('waters.userDetails', JSON.stringify({ ...userDetails, firstName: '', lastName: '', company: '' }));
        enzymeWrapper.setProps({ greetings: 'Welcome!' });
        enzymeWrapper.update();
        expect(enzymeWrapper.find('h2')).toHaveLength(1);
        expect(enzymeWrapper.find('h3')).toHaveLength(1);
        expect(enzymeWrapper.find('h4')).toHaveLength(1);
        expect(enzymeWrapper.find('.logo')).toHaveLength(1);

        expect(enzymeWrapper.find('h2').text()).toBe('Welcome!');
        expect(enzymeWrapper.find('h3').text()).toBe('');
        expect(enzymeWrapper.find('h4').text()).toBe('');
    });
});