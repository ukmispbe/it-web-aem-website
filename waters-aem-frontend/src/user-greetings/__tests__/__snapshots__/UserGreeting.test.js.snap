// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserGreeting /> should render without throwing an error 1`] = `ShallowWrapper {}`;
